### PR TITLE
fix: convert `Symbol` to symbolic variables in remake, fix split_idxs

### PR DESCRIPTION
**Pull request by AayushSabharwal**
## Checklist

- [ ] Appropriate tests were added
- [ ] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [ ] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [ ] Any new documentation only uses public API

### CODE REVIEW

File: src/variables.jl [hunk 1]
```diff
@@ -142,11 +142,19 @@ function SciMLBase.process_p_u0_symbolic(prob::Union{SciMLBase.AbstractDEProblem
         hasproperty(prob.f, :sys) && hasfield(typeof(prob.f.sys), :ps) ||
             throw(ArgumentError("This problem does not support symbolic maps with `remake`, i.e. it does not have a symbolic origin." *
                                 " Please use `remake` with the `p` keyword argument as a vector of values, paying attention to parameter order."))
+        p = [
+            (symbolic_type(sym) != NotSymbolic() ? parameter_symbols(prob)[parameter_index(prob, sym)] : sym) => val
```
Comment (ChrisRackauckas) [inline src/variables.jl:146]: Shouldn't it be the other way around, to always canonicalize to name?

**Comment by ChrisRackauckas** [comment]
What's left here?

**Comment by AayushSabharwal** [comment]
Nothing. This is required for https://github.com/SciML/SciMLBase.jl/pull/583 to pass

**Comment by AayushSabharwal** [comment]
https://github.com/SciML/SciMLBase.jl/pull/583#issuecomment-1922866598

**Pull request closed without merging**